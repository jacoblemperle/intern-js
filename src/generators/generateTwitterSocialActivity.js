import _ from 'lodash';

export default function twitterTransformer (message) {
    const PostMedia = [];
    if (message.images && message.images.standard_resolution) {
        PostMedia.push({
            MediaType: _.get(message, 'type') || 'image',
            URL: message.images.standard_resolution.url,
        });
    }
    const socialActivity = {
        ActivityURL: _.get(message, 'link'),
        AuthorID: _.get(message, 'actor.id'),
        AuthorPictureURL: _.get(message, 'actor.image'),
        AuthorRealName: _.get(message, 'actor.displayName'),
        AuthorUsername: _.get(message, 'actor.preferredUsername'),
        BodyText: _.get(message, 'body'),
        ID: _.get(message, 'id'),
        Network: 'twitter',
        PostMedia,
        Version: '2.0',
    };
    if (socialActivity.ID && socialActivity.BodyText) {
        return socialActivity;
    }
    return undefined;
}
