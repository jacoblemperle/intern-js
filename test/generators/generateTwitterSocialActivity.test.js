import TwitterFixture from '../fixtures/twitter.json';
import TwitterTwoFixture from '../fixtures/twitterTwo.json';
import generateTwitterSocialActivity from '../../src/generators/generateTwitterSocialActivity';

describe('generateTwitterSocialActivity', () => {
  xit('should generate an activity with fixure json', () => {
    const activity = generateTwitterSocialActivity(TwitterFixture);
    expect(activity.ActivityURL).toEqual('http://twitter.com/MckeonLutz/statuses/751152691611463680');
    expect(activity.AuthorID).toEqual('id:twitter.com:1386906962');
    expect(activity.AuthorPictureURL).toEqual('https://abs.twimg.com/sticky/default_profile_images/default_profile_0_normal.png');
    expect(activity.AuthorRealName).toEqual('MckeonLutz');
    expect(activity.AuthorUsername).toEqual('MckeonLutz');
    expect(activity.Network).toEqual('twitter');
    expect(activity.ID).toEqual('tag:search.twitter.com,2005:751152691611463680');
    expect(activity.BodyText).toEqual('RT @StylishRentals: Love this! \"Modern style Country Villa - Houses for Rent in Almoster\" @airbnb #Travelhttps://t.co/xJ479oYcx9');
  });

  xit('should generate an activity with second fixure json', () => {
    const activity = generateTwitterSocialActivity(TwitterTwoFixture);
    expect(activity.ActivityURL).toEqual('http://twitter.com/JenVento/statuses/794169597222064128');
    expect(activity.AuthorID).toEqual('id:twitter.com:18141787');
    expect(activity.AuthorPictureURL).toEqual('https://pbs.twimg.com/profile_images/378800000578979749/4ea77b8e30565b5bf1222b9042ae4e5f_normal.jpeg');
    expect(activity.AuthorRealName).toEqual('Jennifer Vento');
    expect(activity.AuthorUsername).toEqual('JenVento');
    expect(activity.Network).toEqual('twitter');
    expect(activity.ID).toEqual('tag:search.twitter.com,2005:794169597222064128');
    expect(activity.BodyText).toEqual('How women could vote Hillary Clinton into the White House https://t.co/jnFqk44nr4 #imwithher');
  });

  xit('should returned undefined if missing id and body text', () => {
  });

  xit('should match snapshot on valid activity', () => {
  });

});
