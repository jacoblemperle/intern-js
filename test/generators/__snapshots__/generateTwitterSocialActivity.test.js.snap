// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTwitterSocialActivity should match snapshot on valid activity 1`] = `
Object {
  "ActivityURL": "http://twitter.com/MckeonLutz/statuses/751152691611463680",
  "AuthorID": "id:twitter.com:1386906962",
  "AuthorPictureURL": "https://abs.twimg.com/sticky/default_profile_images/default_profile_0_normal.png",
  "AuthorRealName": "MckeonLutz",
  "AuthorUsername": "MckeonLutz",
  "BodyText": "RT @StylishRentals: Love this! \\"Modern style Country Villa - Houses for Rent in Almoster\\" @airbnb #Travelhttps://t.co/xJ479oYcx9",
  "ID": "tag:search.twitter.com,2005:751152691611463680",
  "Network": "twitter",
  "PostMedia": Array [],
  "Version": "2.0",
}
`;
